autoload/neomake/makers/typescript_project.vim

```diff
@@ -0,0 +1,13 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#typescript_project#typescript_project()
+    return {
+        \ 'exe': 'tsc',
+        \ 'args': ['--noEmit'],
+        \ 'errorformat':
+            \ '%E%f %#(%l\,%c): error %m,' .
+            \ '%E%f %#(%l\,%c): %m,' .
+            \ '%Eerror %m,' .
+            \ '%C%\s%\+%m'
+        \ }
+endfunction
```

Comment:
Check out https://github.com/neomake/neomake/pull/502 regarding this: it might still not pass through `bufnr`, but then it's a matter of the hook handling it.